### DISCUSSION TITLE
How to avoid downloads error from Windows 2022 template creation

### DISCUSSION THREAD

**Discussion by nico3133** [actions/virtual-environments#8635] [General questions] [answered]
Hi all,

Trying to generate a template for Windows2022, but, at each attempt it fails on some random error like :
`Found PyPy 'pypy3.9-v7.3.13-win64.zip' package
    azure-arm.image: Downloading package from: https://downloads.python.org/pypy/pypy3.9-v7.3.13-win64.zip to path C:\Users\packer\AppData\Local\Temp\pypy3.9-v7.3.13-win64.zip .
    azure-arm.image: Package downloaded successfully in 0.06 seconds
==> azure-arm.image:   WARNING: The scripts pip.exe, pip3.11.exe, pip3.8.exe and pip3.exe are installed in 'C:\hostedtoolcache\windows\PyPy\3.8.16\x86\Scripts' which is not on PATH.
==> azure-arm.image:   Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
==> azure-arm.image: Checksum verification failed. Expected hash: 09EA41154AD1DCD3E8378609A73196A6C108B17AA05EF3CBB240610744102803; Actual
`

And last time on the mongo db, because name of the release was not OK in the script which download Mongo DB

Is it possible to continue on this errors ? because my goal is to execute automatic nigthly builds of templates, without failures ?

Thanks for you're help, and you're incredible work on this git !!!

Nicolas

**Comment by vpolikarpov-akvelon** [comment, marked as answer]
Hi, @nico3133. Checksum comparison was introduced recently in order to fit some security and compliance requirements. If you don't need it you can alter behavior of corresponding functions. E.g. you may replace `throw` in [Use-ChecksumComparison](https://github.com/actions/runner-images/blob/254e66d02d44090b09327d1682949e7d204ffc2f/images/win/scripts/ImageHelpers/InstallHelpers.ps1#L662-L677) with `Write-Host`. Unfortunately, there is no any conventional way to disable that verification.

Also it's always recommended to build images from main branch and avoid using releases or tags. More on that here: https://github.com/actions/runner-images/discussions/8476#discussioncomment-7209143

**Reply by nico3133** [reply]
Hello @vpolikarpov-akvelon 
THanks for your answer, I understand.

I always use a clone from the main, so I hope all will be good.

Thanks for you quick answer, have a nice day !
